src/hooks/useEmailNotifications.js

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useState } from 'react'
+import { courtServerEndpoint } from '../endpoints'
+
+const COURT_SERVER_ENDPOINT = courtServerEndpoint()
+
+async function notificationServiceCall(
+  method,
+  address,
+  apiModule,
+  body,
+  action
+) {
+  if (!address) {
+    return
+  }
+
+  const endpoint = `${COURT_SERVER_ENDPOINT}/users/${address}${
+    apiModule ? `/${apiModule}` : ''
+  }${action ? `:${action}` : ''} `
+
+  try {
+    const rawResponse = await fetch(endpoint, {
+      method: method,
+      credentials: 'include',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: body && JSON.stringify(body),
+    })
+
+    const jsonResponse = await rawResponse.json()
+
+    if (rawResponse.ok) {
+      return {
+        response: { ...jsonResponse },
+        responseStatus: rawResponse.status,
+        error: false,
+        errorMsg: '',
+      }
+    }
+
+    const errors = jsonResponse.errors
+      .map(err => Object.values(err).join(', '))
+      .join(', ')
+
+    return {
+      responseStatus: rawResponse.status,
+      error: true,
+      errorMsg: errors,
+    }
+  } catch (err) {
+    console.error(err)
+    return { error: true, errorMsg: err }
+  }
+}
+
+export function useSubscriptionDetails(address) {
```

Comment:
We should probably run this as a `useEffect()` in the shared notification context, and be dependent on `useWallet()`'s account param to re-run.